### ISSUE TITLE
Op-reth v1.0.4 broken debug_traceBlockByNumber

### ISSUE THREAD

**Issue by argakiig** [paradigmxyz/reth#10208]
### Describe the bug

rpc method debug_traceBlockByNumber is broken on v1.0.4

v1.0.4
```
// Request
{
    "method": "debug_traceBlockByNumber",
    "params":["0x114a730",{"tracer":"callTracer"}],
    "id": 1,
    "jsonrpc": "2.0"
}
// Response
{
    "jsonrpc": "2.0",
    "id": 1,
    "error": {
        "code": -32003,
        "message": "max fee per gas less than block base fee"
    }
}
```
v1.0.3 and below

![image](https://github.com/user-attachments/assets/6aea2d7e-768b-4f79-a4a4-36b8d3eeb219)

### Steps to reproduce

1) upgrade node from 1.0.0-1.0.3 to 1.0.4 rpc method stops returning valid responses for `debug_traceBlockByNumber`

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth Version: 1.0.4-dev
Commit SHA: e24e4c773d7571a5a54dba7854643c02d0b0a841
Build Timestamp: 2024-08-06T19:26:05.400257125Z
Build Features: asm_keccak,jemalloc,optimism
Build Profile: maxperf

### What database version are you on?

2

### Which chain / network are you on?

base/optimism

### What type of node are you running?

Archive (default)

### If you've built Reth from source, provide the full command you used

cargo build --profile maxperf --features jemalloc,asm-keccak,optimism --locked --bin op-reth

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
possibly closed by https://github.com/paradigmxyz/reth/pull/10210

**Comment by argakiig** [comment]
this is fixed as of 1.0.5. Thank you
